src/solver.rs

```diff
@@ -121,10 +121,15 @@ pub fn resolve<P: Package, V: Version>(
             }
             Some(x) => x,
         };
-        debug_assert!(state
+        if state
             .partial_solution
             .term_intersection_for_package(&next)
-            .contains(&v));
+            .contains(&v)
```

Comment:
shouldn't be the reverse? the version is supposed to be contained in the intersection of terms.